obs-server/check_new_uploads

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+
+import imaplib
+import email
+import subprocess
+import time
+
+TRIGGER_BIN_PATH = '/usr/local/bin/trigger_clang_build'
+
+# Hours to wait before triggering a build;
+# this allows propagation time so the package is available in source mirrors.
+COOLDOWN = 6
+
+# Mail server IMAP information and credentials;
+# this must be an email subscribed to the debian-devel-changes mailing list.
+IMAP_SERVER = "{{ salt['pillar.get']('email-credentials:lookup:hostname') }}"
+USERNAME = "{{ salt['pillar.get']('email-credentials:lookup:user') }}"
+PASSWORD = "{{ salt['pillar.get']('email-credentials:lookup:pass') }}"
+
+
+def parse_message(msg):
+    lines = msg.split('\n')
+    mapping = {
+        'Source': 'name',
+        'Version': 'version',
+        'Architecture': 'arch',
+        'Distribution': 'dist',
+        'Maintainer': 'maintainer'
+    }
+    pkg = {}
+    for line in lines:
+        if len(mapping) == 0:
+            break
+        for field, target in mapping.items():
+            if line.startswith('%s: ' % field):
+                val = line[len(field) + 2:]
+                pkg[target] = val.strip()
+                del mapping[field]
+                break
+    return pkg
+
+
+def trigger_build(pkg):
+    subprocess.run([TRIGGER_BIN_PATH, pkg['name']])
+
+
+subprocess.run(['apt', 'update'])
+list_id = '<debian-devel-changes.lists.debian.org>'
+M = imaplib.IMAP4_SSL(IMAP_SERVER)
+M.login(USERNAME, PASSWORD)
+M.select()
+typ, data = M.search(None, 'UNSEEN')
+for num in data[0].split():
+    typ, data = M.fetch(num, '(RFC822)')
+    parsed_email = email.message_from_bytes(data[0][1])
+
+    # We want to give some time so the upload can propagate to the mirrors
+    email_time = email.utils.parsedate_tz(parsed_email['date'])
+    email_time_utc = email.utils.mktime_tz(email_time)
+    upload_time_limit = time.time() - COOLDOWN*60*60
+    if email_time_utc > upload_time_limit:
+        # Too recent to be parsed
+        M.store(num, '-FLAGS', '\\SEEN')
+        continue
+
+    if(parsed_email['subject'].startswith('Accepted')
+       and parsed_email['list-id'] == list_id):
+        pkg = parse_message(parsed_email.get_payload())
+        if (pkg['dist'] in ['unstable', 'sid']) and 'source' in pkg['arch']:
+            trigger_build(pkg)
+    # Uncomment the following lines if you want to remove the email afterwards
+    # M.store(num, '+FLAGS', '\\Deleted')
```

Comment:
why we are not doing that? (removing the email)

Reply:
No specific reason. I Just uncommented the lines :)